File: 2024/dec01/1.py
#!/usr/bin/python3
from aoc import *
sub = lambda a, b: a-b
diff = lambda a, b: abs(sub(a, b))
data = list(map(str.split, stdin.readlines()))
a = sorted([int(a[0]) for a in data])
b = sorted([int(b[1]) for b in data])
print(sum(starmap(diff, zip(a, b))))

File: 2024/dec01/1.pl
#!/usr/bin/perl
use 5.020;
use List::Util qw(sum);
use List::MoreUtils qw(zip6);
my (@a, @b);
for my $pair (map { [map { int } split / +/] } <>) {
	push @a, $pair->[0];
	push @b, $pair->[1];
}
@a = sort @a;
@b = sort @b;
say sum(map { abs $_->[0] - $_->[1] } zip6(@a, @b));
